internal/transform/scope-html.go

```diff
@@ -39,37 +41,38 @@ var NeverScopedSelectors map[string]bool = map[string]bool{
 func injectScopedClass(n *astro.Node, opts TransformOptions) {
 	for i, attr := range n.Attr {
 		// If we find an existing class attribute, append the scoped class
-		if attr.Key == "class" || (n.Component && attr.Key == "className") {
+		if attr.Key == "data-astro-scope" {
 			switch attr.Type {
 			case astro.ShorthandAttribute:
 				if n.Component {
-					attr.Val = attr.Key + ` + " astro-` + opts.Scope + `"`
+					attr.Val = fmt.Sprintf(`%s + "%s"`, attr.Key, opts.Scope)
 					attr.Type = astro.ExpressionAttribute
 					n.Attr[i] = attr
 					return
 				}
 			case astro.EmptyAttribute:
 				// instead of an empty string
 				attr.Type = astro.QuotedAttribute
-				attr.Val = "astro-" + opts.Scope
+				attr.Val = opts.Scope
 				n.Attr[i] = attr
 				return
 			case astro.QuotedAttribute, astro.TemplateLiteralAttribute:
 				// as a plain string
-				attr.Val = attr.Val + " astro-" + opts.Scope
+				attr.Val = fmt.Sprintf(`%s %s`, attr.Val, opts.Scope)
 				n.Attr[i] = attr
 				return
 			case astro.ExpressionAttribute:
 				// as an expression
-				attr.Val = "(" + attr.Val + `) + " astro-` + opts.Scope + `"`
+				attr.Val = fmt.Sprintf(`(%s) + " %s"`, attr.Val, opts.Scope)
 				n.Attr[i] = attr
 				return
 			}
 		}
 	}
 	// If we didn't find an existing class attribute, let's add one
 	n.Attr = append(n.Attr, astro.Attribute{
-		Key: "class",
-		Val: "astro-" + opts.Scope,
+		Key:  "data-astro-scope",
+		Type: astro.QuotedAttribute,
```

Comment:
Just curious, why switch to a data attribute instead of a class name?

Reply:
Great question! It's not strictly necessary since `:where()` will reset both `.astro-XXXXXX` and `data-astro-scope="XXXXXX"` to zero.

The motivation is that this attribute would be much more explicit. Currently `class` has an odd behavior where we append the parent `.astro-XXXXXX` to it. If you spread `{...Astro.props}` onto the child, it will inherit that `.astro-XXXXXX` and **pierce scoped encapsulation** without users realizing it. Separating this out to a `data-` attribute means we can filter it from `props` and allow users to opt-in to piercing behavior with a directive. It also removes the special-casing of `className` on component nodes and matches Vue's approach (which uses an attribute like `data-v-469af010`).

Reply:
I switched this back to our original approach with `class`! Hopefully makes this a smaller change to get in for v1.0.0-rc.